pylint/test/functional/mutable_sequence_modified_in_loop.py

```diff
@@ -0,0 +1,66 @@
+# coding: utf-8
+""" The mutable sequence was modified inside the loop """
+
+# pylint:disable=print-statement, missing-docstring, redefined-outer-name, invalid-name, using-constant-test, no-member
+
+
+class MyList(object):
+
+    def __init__(self):
+        self.my_list = range(1, 6)
+
+    def __iter__(self):
+        return iter(self.my_list)
+
+    def remove(self, item):
+        pass
+
+    def delete(self, item):
+        pass
+
+    def pop(self, item):
+        pass
+
+
+my_list = MyList()
+for item in my_list:
+    my_list.remove(item)
+
+my_list = MyList()
+for item in my_list:
+    my_list.delete(item)
+
+my_list = MyList()
+for item in my_list:
+    my_list.pop(item)
+
+
+items = range(1, 6)
```

Comment:
Is really necessary separated the test case on Python3 and Python2?
The last travis build on py36 is green https://travis-ci.org/PyCQA/pylint/jobs/199682219 and py27 is green https://travis-ci.org/PyCQA/pylint/jobs/199682214

On this line https://github.com/PyCQA/pylint/pull/1297/files#diff-a4a630f96d773cb3cbba7b8663916622R295 Is valid when it is Python3